src/test/java/uk/gov/hmcts/reform/iacaseapi/domain/service/ApplyForCostsAppenderTest.java

```diff
@@ -0,0 +1,238 @@
+    @Mock
+    private UserDetailsHelper userDetailsHelper;
+    @Mock
+    private DateProvider dateProvider;
+    @Mock
+    private IdValue<ApplyForCosts> applyForCostsById1;
+    @Mock
+    private IdValue<ApplyForCosts> applyForCostsById2;
+
+    private TypesOfAppliedCosts typesOfAppliedCosts = TypesOfAppliedCosts.UNREASONABLE_COSTS;
+    private TypesOfAppliedCosts typesOfAppliedCostsDesc = TypesOfAppliedCosts.WASTED_COSTS;
+    private String argumentsAndEvidenceDetails = "Test details";
+
+    private List<IdValue<Document>> argumentsAndEvidenceDocuments =
+            List.of(new IdValue<>("1",
+                    new Document("http://localhost/documents/123456",
+                            "http://localhost/documents/123456",
+                            "DocumentName.pdf")));
+    private List<IdValue<Document>> scheduleOfCostsDocuments = Collections.emptyList();
+    private YesOrNo applyForCostsHearingType = YesOrNo.YES;
+    private String applyForCostsHearingTypeExplanation = "Test explanation";
+    private String applyForCostsDecision = "Test decision";
+    private String applyForCostsApplicantType = "Respondent";
+    private String applyForCostsApplicantTypeLegalRep = "Legal representative";
+    private String respondentToCostsOrder = "Legal representative";
+    private String respondentToCostsOrderHomeOffice = "Home office";
+    private String applyForCostsCreationDate = "2020-09-21";
+    private ApplyForCostsAppender applyForCostsAppender;
+
+    @BeforeEach
+    public void setUp() {
+        applyForCostsAppender =
+                new ApplyForCostsAppender(userDetails, userDetailsHelper, dateProvider);
+    }
+
+    @Test
+    void should_append_the_new_apply_for_costs_app_in_first_position() {
+
+        when(dateProvider.now()).thenReturn(LocalDate.of(2020, 9, 21));
+        when(userDetailsHelper.getLoggedInUserRoleLabel(userDetails)).thenReturn(UserRoleLabel.HOME_OFFICE_GENERIC);
+
+        ApplyForCosts existingApplyForCosts1 = mock(ApplyForCosts.class);
+        when(applyForCostsById1.getValue()).thenReturn(existingApplyForCosts1);
+
+        ApplyForCosts existingApplyForCosts2 = mock(ApplyForCosts.class);
+        when(applyForCostsById2.getValue()).thenReturn(existingApplyForCosts2);
+
+        List<IdValue<ApplyForCosts>> existingAppliesForCosts = List.of(applyForCostsById1, applyForCostsById2);
+
+        List<IdValue<ApplyForCosts>> allAppliesForCosts = applyForCostsAppender.append(
+                existingAppliesForCosts,
+                typesOfAppliedCosts,
+                argumentsAndEvidenceDetails,
+                argumentsAndEvidenceDocuments,
+                scheduleOfCostsDocuments,
+                applyForCostsHearingType,
+                applyForCostsHearingTypeExplanation,
+                applyForCostsDecision);
+
+        assertNotNull(allAppliesForCosts);
+        assertEquals(3, allAppliesForCosts.size());
+
+        assertEquals("3", allAppliesForCosts.get(0).getId());
```

Comment:
How is the algorithm working for the Ids?

Reply:
ExUi is not showing as the opposite order(in java it is sorted but not in exui)